osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -249,31 +249,32 @@ private class ChannelScrollContainer : UserTrackingScrollContainer
             /// </summary>
             private const float auto_scroll_leniency = 10f;
 
+            private bool trackNewContent = true;
             private float? lastExtent;
 
-            protected override void OnUserScroll(float value, bool animated = true, double? distanceDecay = default)
+            protected override void OnScrollChange(bool byUser)
             {
-                base.OnUserScroll(value, animated, distanceDecay);
-                lastExtent = null;
+                base.OnScrollChange(byUser);
+
+                if (byUser)
+                    lastExtent = null;
+
+                trackNewContent = IsScrolledToEnd(auto_scroll_leniency);
             }
 
             protected override void Update()
             {
                 base.Update();
 
-                // If the user has scrolled to the bottom of the container, we should resume tracking new content.
-                if (UserScrolling && IsScrolledToEnd(auto_scroll_leniency))
-                    CancelUserScroll();
-
                 // If the user hasn't overridden our behaviour and there has been new content added to the container, we should update our scroll position to track it.
-                bool requiresScrollUpdate = !UserScrolling && (lastExtent == null || Precision.AlmostBigger(ScrollableExtent, lastExtent.Value));
+                bool requiresScrollUpdate = trackNewContent && (lastExtent == null || Precision.AlmostBigger(ScrollableExtent, lastExtent.Value));
```

Comment:
So I'm very greatly confused here. Why is any of this refactoring needed if the *only* thing that matters is `trackNewContent`, which is just an `IsScrolledToEnd()` check? As in, given that this is in `Update()`, shouldn't I be able to replace all of this by just checking `IsScrolledToEnd()` every frame, at which point `OnScrollChange()` has zero reason to exist?

Reply:
Well, that can't exactly work, `IsScrolledToEnd` will no longer pass when the extent of the scroll container changes, `trackNewContent` stores the value of that from the latest scroll operation.

Reply:
Ok, but that doesn't change the fact that the value of `UserScrolling` no longer matters. The only thing that looks to matter is whether chat was scrolled to end in the previous frame. Which, I argue, still renders this refactoring mostly moot - as in you *could* do it, but it's not required to get the behaviour you want.

Reply:
> The only thing that looks to matter is whether chat was scrolled to end in the previous frame.

I can't think of it working that way either, since you would have to, at the current frame, differentiate between "chat was scrolled up because a new message has posted" and "chat was scrolled up because it was done programmatically or by user". 

You could use `ScrollableExtent > lastExtent.Value` as an indicator, but that wouldn't handle the case where chat is scrolled up (programmatically or by user) *and* a message was posted.

Reply:
What I said above actually isn't correct, as it can be easily differentiated based on the state of the scroll container from last frame.

But after thinking about it, even if `ChannelScrollContainer` kept track of whether it was scrolled to end last frame, the problem is, what if a scroll operation was performed in between the "after last frame" and "before current frame" window, it'll be dismissed if the container was scrolled to end last frame.